test/performance/range/view/view_kmer_hash_benchmark.cpp

```diff
@@ -0,0 +1,77 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/range/view/kmer_hash.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/performance/std_kmer_hash.hpp>
+#include <seqan3/test/performance/units.hpp>
+
+using namespace seqan3;
+
+inline benchmark::Counter bp_per_second(size_t const basepairs)
+{
+    return benchmark::Counter(basepairs,
+                              benchmark::Counter::kIsIterationInvariantRate,
+                              benchmark::Counter::OneK::kIs1000);
+}
+
+static void arguments(benchmark::internal::Benchmark* b)
+{
+    for (int32_t sequence_length : {1'000, 50'000, /*1'000'000*/})
+    {
+        for (int32_t k : {8, /*16, 24,*/ 32})
+        {
+            b->Args({sequence_length, k});
+        }
+    }
+}
+
+static void seqan_kmer_hash(benchmark::State & state)
+{
+    auto sequence_length = state.range(0);
+    assert(sequence_length > 0);
+    size_t k = static_cast<size_t>(state.range(1));
+    assert(k > 0);
+    auto seq = test::generate_sequence<dna4>(sequence_length, 0, 0);
+
+    volatile size_t sum{0};
+
+    for (auto _ : state)
+    {
+        for (auto h : seq | view::kmer_hash(k))
+            benchmark::DoNotOptimize(sum += h);
+    }
+
+    state.counters["Throughput[bp/s]"] = bp_per_second(sequence_length - k + 1);
+}
+
+static void std_kmer_hash(benchmark::State & state)
+{
+    auto sequence_length = state.range(0);
+    assert(sequence_length > 0);
+    size_t k = static_cast<size_t>(state.range(1));
+    assert(k > 0);
+    auto seq = test::generate_sequence<dna4>(sequence_length, 0, 0);
+
+    volatile size_t sum{0};
+
+    for (auto _ : state)
+    {
+        for (auto h : seq | view::std_kmer_hash(k))
+            benchmark::DoNotOptimize(sum += h);
+    }
+
+    state.counters["Throughput[bp/s]"] = bp_per_second(sequence_length - k + 1);
+}
+
+BENCHMARK(seqan_kmer_hash)->Apply(arguments);
+BENCHMARK(std_kmer_hash)->Apply(arguments);
+
```

Comment:
Hm, are "seqan_kmer_hash" and "std_kmer_hash" good names? Both are seqan, only one of them is rolling, or not?

Reply:
Not in general, just for ungapped shapes.

Reply:
> Hm, are "seqan_kmer_hash" and "std_kmer_hash" good names? Both are seqan, only one of them is rolling, or not?

well the view is both seqan but the hashing is done with std::hash for std_kmer_hash

Reply:
I still think it's confusing. `std_kmer_hash` looks like its was referrring to `std::kmer::hash` or something else in `std::`....

Reply:
Renamed to `naive_kmer_hash`